colin-api/src/colin_api/models/filing.py

```diff
@@ -1396,11 +1397,19 @@ def add_filing(cls, con, filing: Filing) -> int:
                                                 Business.TypeCodes.BCOMP_CONTINUE_IN.value,
                                             ])
 
-            # Freeze all entities except CP if 'enable-bc-ccc-ulc' flag is on else just freeze BEN
-            is_frozen_condition = (
-                flags.is_on('enable-bc-ccc-ulc') and
-                business['business']['legalType'] != Business.TypeCodes.COOP.value
-            )
+            # default value set to False
+            is_frozen_condition = False
+
+            # call to legal-api to check if business exists in lear
+            response = LegalApiService.query_business(bc_identifier)
+
+            # Check business exists in lear
+            # If found in LEAR, freeze entities except CP if 'enable-bc-ccc-ulc' flag is on else just freeze BEN
+            if response.status_code == HTTPStatus.OK:
```

Comment:
i think you should create a separate function called `is_business_in_lear` that wraps the check and returns a bool.

And add the result of `is_business_in_lear` as one of the conditions in `is_frozen_condition`. 

Think it will be cleaner this way.